go-controller/pkg/node/udn_isolation.go

```diff
@@ -293,52 +295,77 @@ func (m *UDNHostIsolationManager) updateKubeletCgroup() error {
 // If a new cgroup is created, you load the filtering policy for the new cgroup and then add
 // processes to that cgroup. You only have to follow the right sequence to avoid problems.
 func (m *UDNHostIsolationManager) runKubeletRestartTracker(ctx context.Context) (err error) {
-	conn, err := dbus.NewSystemdConnectionContext(ctx)
+
+	conn, err := dbus.Dial("unix:path=/run/systemd/private", dbus.WithContext(ctx))
 	if err != nil {
-		return fmt.Errorf("failed to connect to systemd: %w", err)
+		return err
 	}
+
 	defer func() {
 		if err != nil {
-			conn.Close()
+			if err := conn.Close(); err != nil {
+				klog.Errorf("Error closing dbus connection for UDN isolation: %v", err)
+			}
 		}
 	}()
 
-	err = conn.Subscribe()
+	// Only use EXTERNAL method, and hardcode the uid (not username)
+	// to avoid a username lookup (which requires a dynamically linked
+	// libc)
+	methods := []dbus.Auth{dbus.AuthExternal(strconv.Itoa(os.Getuid()))}
+
+	err = conn.Auth(methods)
 	if err != nil {
-		return fmt.Errorf("failed to subscribe to systemd events: %w", err)
+		return err
 	}
-	// interval is important here as we need to catch the restart state, before it is running again
-	events, errChan := conn.SubscribeUnitsCustom(50*time.Millisecond, 0, func(u1, u2 *dbus.UnitStatus) bool { return *u1 != *u2 },
-		func(s string) bool {
-			return s != "kubelet.service"
-		})
+
+	signalChan := make(chan *dbus.Signal, 10)
```

Comment:
every event that happens when the channel is full will spin up a goroutine, so considering I have got 40 events per second on a regular kind cluster, it may make sense to increase this number to 50

Reply:
I'll bump it to 100, which looks like waht coreos was using anyway